packages/ui/src/lib/Button/Button.tsx

```diff
@@ -1,15 +1,112 @@
+export type ButtonProps = React.ButtonHTMLAttributes<HTMLButtonElement> & {
+  /** Defaults to 'primary' */
+  variant?: ButtonVariant;
+  /** If the button is being used for a 'destructive' action such as cancelling or deleting */
+  destructive?: boolean;
+  /** If this is an icon button the glyph is needed to depict icon */
+  fullwidth?: boolean; // set width to 100%
+  /** increase font-size and padding */
+  large?: boolean;
+  /** If the button is in a loadig state */
+  loading?: boolean;
+  /** The current language */
+  language?: SupportedLocales;
+};
 
-console.log(props)
+const LOADING_ICONS = {
+  destructive: 'button-loading-destructive',
+  primary: 'button-loading',
+  secondary: 'button-loading-secondary',
+  secondaryGray: 'button-loading-secondaryGray',
+  tertiary: 'button-loading-tertiary',
+};
 
-  return (
-    <button className={styles['container']}  {...props}>{children}</button>
-  );
-}
+/**
+ * Clover styled button.
+ * Supports all props that can be applied to a native HTML `button` element. Unlike the native HTML `button` element
+ * the `type` is `button` as default, so `submit` must be specified as the type explicitly when using for form submission.
+ *
+ * MDN Button docs: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#Attributes
+ */
+/**
+ * @title
+ *  Button
+ * @label
+ *  Button
+ * @shortdescription
+ *  Clover styled button.
+ * @keywords
+ *  [Button,Clover button,Secondary, Tertiary, primary button]
+ * @description
+ * Clover styled button. Note: Use sentence case for the label!
+ * Supports all props that can be applied to a native HTML `button` element. Unlike the native HTML `button` element
+ * the `type` is `button` as default, so `submit` must be specified as the type explicitly when using for form submission.
+ * MDN Button docs: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/button#Attributes
+ * @componentUsage
+ * <Button variant={'secondary'} {...args} />
+ * @example
+ * <ButtonExample children={'Click me'}/>
+ * @visibility
+ * false
+ */
+export const Button = forwardRef<HTMLButtonElement, ButtonProps>(
+  (
+    {
+      type = 'button',
+      variant = 'primary',
+      destructive,
+      fullwidth,
+      large,
+      loading,
+      className,
+      children,
+      language = getCurrentLocale(),
+      ...passThroughProps
+    },
+    ref
+  ) => {
+    const text = getLanguageStrings(language);
+    return (
+      <>
+        <button
+          type={type}
+          ref={ref}
+          aria-disabled={loading}
+          className={cx(
+            styles.button,
+            styles[variant],
+            destructive && styles.destructive,
+            fullwidth && styles.fullwidth,
+            large && styles.large,
+            styles.dots,
+            className
+          )}
+          {...passThroughProps}
+        >
+          {loading && (
+            <span className={cx(styles.busyState, destructive && styles.destructive)}>
```

Comment:
this prob could be icon/svg file on its own for readability

Reply:
yeah, i modified this, we use an icon lib, but i wanted to limit deps until something is working.